### DISCUSSION TITLE
How to fix the problem of post-process settings not showing up

### DISCUSSION THREAD

**Discussion by noonka** [Adriwin06/Ultimate-CommonUI-Menu-System#21] [Q&A] [answered]
To see the missing options, you need to click on the down arrow to show more options. Once a certain number of options have been selected, they become hidden unless you click on this area of the node "MakePostProcessSettings".

![Captura de pantalla 2024-10-23 102848](https://github.com/user-attachments/assets/f500b13f-2391-4b7c-90d8-e8ec294ad2e6)

Another thing, is there a way to know if the user has a graphics card compatible with hardware ray tracing in order to hide or show the graphical options, similar to how it’s done with DLSS?

PS: Congrats for your great work

**Comment by Adriwin06** [comment]
> To see the missing options, you need to click on the down arrow to show more options. Once a certain number of options have been selected, they become hidden unless you click on this area of the node "MakePostProcessSettings".
> 
Bruh it was only that 😭🤣
You can make a pull request since you are the one who find the problem is you want :)

> Another thing, is there a way to know if the user has a graphics card compatible with hardware ray tracing in order to hide or show the graphical options, similar to how it’s done with DLSS?
> 
I searched everywhere for this, but the only way I found to get informations like that about the GPU is trough a paid plugin on the Marketplace, so I can't add it in this open source project...

> PS: Congrats for your great work
>
Thanks 😊

**Reply by noonka** [reply]
> You can make a pull request since you are the one who find the problem is you want :) <

nah, it is not necessary, but thanks.

> I searched everywhere for this, but the only way I found to get informations like that about the GPU is trough a paid plugin on the Marketplace, so I can't add it in this open source project... <

I've been looking through the free plugins and maybe you can do something half right with this 
[Extra Win Functions](https://www.fab.com/listings/dbf64b8c-4d92-400f-ab8f-a60f5f7c4d83)
Had this nodes:
**Get all Graphics Card names** and **Get Primary Graphics Card Name**
But i don't tried it yet

**Reply by Adriwin06** [reply]
I will try as soon as I can download it... Fab is so great that we can't even download what we acquired on it, since the Vault on the Epic Games launcher doesn't works, and that there is currently not the plugin for UE5...

**Reply by noonka** [reply]
yeah, the launch has been disastrous, on the bright side, there is only room for improvement from here LOL

**Reply by Adriwin06** [reply]
You are right, even if they wanted to, it would be impossible to do worse XD

**Reply by noonka** [reply]
I tried the plugin and look like it's working, the node **Get all Graphics Card names**  return the integrated graphic card and the RTX, the node **Get Primary Graphic Card** return my active graphic card

> LogBlueprintUserMessages: [WBP_AdvGraphicsSettings_C_0] NVIDIA GeForce RTX 4060 Ti

So a **Contains** String node with a "RTX" in the substring you should be able to detect if it is an RTX graphics card. I'm sure there's a better way, but this is the first one that came to my mind.

![Captura de pantalla 2024-10-23 174520](https://github.com/user-attachments/assets/c4d03cf8-e04d-449f-aa98-bd7386ce694c)

**Reply by Adriwin06** [reply]
Yeah but AMD and Intel GPU can ray Trace too, and their name don't start with RTX, and for AMD, it star with RX, but only RX 6000 or higher can

**Reply by noonka** [reply]
ah, I thought ray tracing was a propietary thing for NVIDIA -.-'

**Reply by Adriwin06** [reply]
Well Nvidia where the first to introduce it, but then AMD did the same, and then Intel

**Reply by noonka** [reply]
well, maybe an array with all the models compatible and checking it with a For Each Loop With Break, but it can be a bit cumbersome to maintain and you have to be sure the name that each model will display in the node output, but ey, it's free lol

**Reply by Adriwin06** [reply]
Ok it showed up in my vault, nice

**Reply by Adriwin06** [reply]
and yeah good idea the array, but it will need manual update for every new GPU...

**Reply by noonka** [reply]
i'm making my own framework and i have a data asset where i configure a lot of things, i can add the array to it and make it a little more easy to update, but if i don't update the game with each new model, yeah, it's a problem

**Reply by Adriwin06** [reply, marked as answer]
I guess it's time to start using C++ and do something like that:
```cpp
#include "RHI.h"
#include "RHIDefinitions.h"
#include "YourProjectName.h" // Project's header file
#include "RayTracingSupportHelper.generated.h"

UCLASS()
class PROJECTNAME_API URayTracingSupportHelper : public UObject
{
    GENERATED_BODY()

public:
    UFUNCTION(BlueprintCallable, Category = "Ray Tracing")
    static bool IsRayTracingSupported()
    {
        // Check if the GPU supports Ray Tracing
        return GSupportsRayTracing;
    }
};
```

**Reply by noonka** [reply]
Yeah, its look more promising
